src/boundary_conditions/boundary_conditions.cpp

```diff
@@ -267,9 +267,17 @@ void incflo::init_bcs ()
             else if (bct == BC::slip_wall)
             {
                 if (side == Orientation::low) {
-                    m_bcrec_density[0].setLo(dir, BCType::hoextrap);
+                    if (m_advection_type == "BDS") {
+                        m_bcrec_density[0].setLo(dir, BCType::foextrap);
+                    } else{
+                        m_bcrec_density[0].setLo(dir, BCType::hoextrap);
```

Comment:
Can you apply this to scalars and tracers as well, and leave a comment "// BDS requires foextrap to avoid introduction of local max/min"